.travis.yml

```diff
@@ -18,16 +17,26 @@ env:
   global:
     - RADICAL_PILOT_DBURL=mongodb://rct:rct_test@one.radical-project.org/rct_test
     - RADICAL_PILOT_TEST_DBNAME=rct_test
+    - LOG=`git log -n 1 | grep Merge`
+    - OLD=`echo $LOG | cut -d ' ' -f2`
+    - NEW=`echo $LOG | cut -d ' ' -f3`
+    - DIFF=`git diff --name-only --diff-filter=b $OLD...$NEW`
+    - DIFF=$(echo $DIFF | grep -o -e '\b[^ ]*.py\b')
     - LOC=/home/travis/virtualenv/python2.7  # Location where VE is created on travis
-    - CMD1="coverage run -m pytest -vvv tests/;flake8 tests/;pylint tests/"
-    - CMD2="flake8 --config=.flake8 $LOC/lib/python2.7/site-packages/radical/pilot/agent/lm/;flake8 --config=.flake8 $LOC/lib/python2.7/site-packages/radical/pilot/agent/rm/;"
-    - CMD3="pylint $LOC/lib/python2.7/site-packages/radical/pilot/agent/lm/;pylint $LOC/lib/python2.7/site-packages/radical/pilot/agent/rm/"
+    - CMD_PYTEST="coverage run -m pytest -vvv tests/"
+    - CMD_FLAKE8="if ! test -z \"$DIFF\"; then flake8 --config=.flake8 $DIFF;else echo 'Nothing to pep8'; fi"
+    - CMD_PYLINT="if ! test -z \"$DIFF\"; then pylint $DIFF;else echo 'Nothing to lint';fi"
+    - CMD_PYTEST="coverage run -m pytest -vvv tests/;flake8 tests/;pylint tests/"
```

Comment:
Why is this redefining `CMD_PYTEST` from above?  It also duplicates the test specific linters below I believe?

Reply:
crap... This should be removed. Thank you for catching it!